### DISCUSSION TITLE
Trouble fetching REST API data

### DISCUSSION THREAD

**Discussion by mike-420** [nuxt/nuxt#10857] [Questions (Nuxt 2)] [answered]
I've been able to fetch data successfully using code such as the following.
``
```
<template>
<div>
  <div>API Test</div>
  <div>
    <ul>
      <li v-for="user in users" :key="user.id">
        {{ user.name }}
      </li>
    </ul>
  </div>  
</div>
</template>
```

```
<script setup>
  const { data: users } = await useAsyncData('users2', () => $fetch('https://jsonplaceholder.typicode.com/users'))
</script>
```

But when I try using my own following API for a list of stores (listed in the API as shops) it doesn't work.
https://mocki.io/v1/3fa1924f-6c6b-4d49-b9ec-d91f6da13c3c

**Can someone please tell me how I should adjust the following code to display all the names of the shops in the Mocki API above?** 

```
<li v-for="user in users" :key="user.id">
        {{ user.name }}
</li>
```

and 

```
<script setup>
  const { data: users } = await useAsyncData('users2', () => $fetch('https://jsonplaceholder.typicode.com/users'))
</script>
```

**I assumed it was the following but it doesn't work.**

```
<li v-for="shop in shops" :key="shop.id">
        {{ shop.name }}
</li>
```

and 

```
<script setup>
  const { data: shops } = await useAsyncData('users2', () => $fetch('https://mocki.io/v1/3fa1924f-6c6b-4d49-b9ec-d91f6da13c3c'))
</script>
```

**Comment by ndragun92** [comment, marked as answer]
This is an example implementation that I have made for you

https://stackblitz.com/edit/github-uvmqdk?file=app.vue

Hope it helps and solved your issue

**Reply by mike-420** [reply]
Thank you ndragun92! **You are awesome!** I had given up trying to learn Nuxt after not being able to get passed successfully connecting to the database but now I'm motivated to keep trying. I can't thank you enough for the example you provided. :)
